assets/pages/monthly-contributions/reducers/reducers.js

```diff
@@ -8,16 +8,47 @@ import stripeCheckout from 'helpers/stripeCheckout/stripeCheckoutReducer';
 import type { Action } from '../actions/monthlyContributionsActions';
 
 
+// ----- Types ----- //
+
+export type State = {
+  amount: number,
+  country: string,
+  firstName: ?string,
+  lastName: ?string,
+  error: ?string,
+};
+
+
+// ----- Setup ----- //
+
+const initialState: State = {
+  amount: 5,
+  country: 'GB',
+  firstName: null,
+  lastName: null,
+  error: null,
+};
+
+
 // ----- Reducers ----- //
 
 function monthlyContrib(
-  state: number = 5,
-  action: Action): number {
+  state: State = initialState,
+  action: Action): State {
 
   switch (action.type) {
 
     case 'SET_CONTRIB_VALUE':
-      return action.value;
+      return Object.assign({}, state, { amount: action.value });
+
+    case 'SET_FIRST_NAME':
+      return Object.assign({}, state, { firstName: action.name });
```

Comment:
Since we will have different forms that will accept first name and last name, I believe it would be better to extract the user logic from here and add it to a user reducer.